fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java

```diff
@@ -214,11 +219,58 @@ protected Response getContent(final String rangeValue,
         return ok(outputStream).build();
     }
 
+
+    protected URI getURLAccessType(final MediaType mediaType) {
+        return URI.create(getAccessTypeValue(mediaType.getParameters(), URL_ACCESS_TYPE));
```

Comment:
`getAccessTypeValue()` seems to only get called here and can be replaced with a stream. So maybe this function should just be
```
protected URI getURLAccessType(final MediaType mediaType) {
    final String accessType = mediaType.getParameters().entrySet().stream()
            .filter(t -> t.getKey().equalsIgnoreCase(URL_ACCESS_TYPE)).map(Map.Entry::getValue)
            .findFirst().orElse(null);
    return URI.create(accessType);
}
```
However `URI.create(null)` is probably not helpful, so maybe we should throw an Exception there?

Also a unit test of this function might be helpful?

Reply:
good points @whikloj.  I'll address as part of the refactoring requested below.